### PR TITLE
Add info hints to new project, add member

### LINKED ISSUE #4217: Registry: adding a member is confusing for regular users
Since non cluster admins cannot list users, the user/group drop-down menu is not populated. It's not clear one can enter text into this field. It would help if the drop-down help text was something like "Select or type user or group"

### PR DESCRIPTION
**Pull request by aweiteka**
Resolves #4217, #4216, #4187, #4188 

**Add member tooltip:**
add_member

**manage image access hyperlink:**
image_sharing

**New project button** ( resolving #4188)
new-project

**new project tool tip:**
popover

### CODE REVIEW

File: pkg/kubernetes/views/image-listing.html [hunk 1]
```diff
@@ -7,6 +7,10 @@
                     <span translatable="yes">New image stream</span>
                 </a>
                 <h3 translatable="yes">Images</h3>
+                <a ng-href="#{{ viewUrl('projects') }}" translatable="yes">
```
Comment (stefwalter) [inline pkg/kubernetes/views/image-listing.html]: This shouldn't have translatable.

File: pkg/kubernetes/views/image-listing.html [hunk 2]
```diff
@@ -7,6 +7,10 @@
...
+                  <i translatable="yes" class="pficon pficon-users"></i>
```
Comment (stefwalter) [inline pkg/kubernetes/views/image-listing.html]: This shouldn't have translatable.

File: pkg/kubernetes/views/project-modify.html [hunk 3]
```diff
@@ -55,6 +55,13 @@ <h3 class="modal-title" ng-if="project" translatable="yes">Change project</h3>
                 </td>
             </tr>
         </table>
+        <div class="alert dialog alert-info" ng-if="!project">
+          <span class="pficon pficon-info"></span>
+          <span translatable="yes">
+            A project is a namespace for images. It corresponds to a Docker repository.
```
Comment (stefwalter) [inline pkg/kubernetes/views/project-modify.html]: I don't think a project corresponds to a docker repository. An imagestream responds to a docker repository.

**Comment by stefwalter** [comment]
Nice work. A had a few review points.

Also, have you seen the 'Access Policy' link here? 

https://raw.githubusercontent.com/cockpit-project/cockpit-design/master/kubernetes/images.png

Do you think it works for the 'Manage image access' case? In particular it should link directly to the relevant project. Instead of just to the list of projects and then expect the user to know what to do next.

**Comment by stefwalter** [comment]
@andreasn Do you have any ideas on how to do this better? Perhaps with a popover? https://www.patternfly.org/widgets/#popover

screenshot from 2016-04-22 21-18-59

File: pkg/kubernetes/views/project-modify.html (hunk 3 above)
Comment (aweiteka) [inline pkg/kubernetes/views/project-modify.html]: What I'm trying to show is how this value is used. It becomes a key part of how the images are addressed. How about something like:

"A project is a namespace for images. It is used as the repository name when referencing images, registry/**project**/image:[tag]."

Comment (stefwalter) [inline pkg/kubernetes/views/project-modify.html]: That's better.

**Comment by aweiteka** [comment]
> Do you think it works for the 'Manage image access' case? In particular it should link directly to the relevant project. Instead of just to the list of projects and then expect the user to know what to do next.

@stefwalter yes, I think this is better. I can update.

**Comment by andreasn** [comment]
For some reason, I get stuck on this when I run make:
GEN      pkg/kubernetes/kubernetes-templates.js

No errors or anything. It just gets stuck.

**Comment by aweiteka** [comment]
@stefwalter @andreasn please review. Should be all set now. In the future I'll break up commits into separate PRs.

File: pkg/kubernetes/views/add-member-role-dialog.html [hunk 4]
```diff
@@ -1,6 +1,12 @@
 <modal-dialog>
     <div class="modal-header">
-        <h4 class="modal-title" translatable="yes">Add Member</h4>
+        <h4 class="modal-title" translatable="yes">
```
Comment (petervo) [inline pkg/kubernetes/views/add-member-role-dialog.html]: Unfortunately translatable doesn't work on elements that have children. For now we proabably need to wrap just the "Add Member" part in a span and only have the translatable there.

File: pkg/kubernetes/views/image-listing.html [hunk 5]
```diff
@@ -64,3 +64,4 @@ <h3 translatable="yes">Images</h3>
         </tr>
     </thead>
 </table>
+
```
Comment (petervo) [inline pkg/kubernetes/views/image-listing.html]: whitespace change

File: pkg/kubernetes/views/imagestream-body.html [hunk 6]
```diff
@@ -1,4 +1,14 @@
 <dl>
+    <dt translatable="yes">Access Policy</dt>
+    <dd translatable="yes">
+      <div ng-if="sharedImages(stream.metadata.namespace)" >
+        <a ng-href="#/projects/{{ stream.metadata.namespace }}">Images shared with authenticated non-members</a>
```
Comment (petervo) [inline pkg/kubernetes/views/imagestream-body.html]: translatable should be on the a not the dd

File: pkg/kubernetes/views/imagestream-body.html [hunk 7]
```diff
@@ -1,4 +1,14 @@
...
+        <i title="Images shared with non-members" class="fa fa-unlock imagestream-lock"></i>
+      </div>
+      <div ng-if="!sharedImages(stream.metadata.namespace)">
+        <a ng-href="#/projects/{{ stream.metadata.namespace }}">Images only accessible to members</a>
```
Comment (petervo) [inline pkg/kubernetes/views/imagestream-body.html]: translatable

File: pkg/kubernetes/views/project-modify.html [hunk 8]
```diff
@@ -1,13 +1,23 @@
 <modal-dialog>
     <div class="modal-header">
-        <h4 class="modal-title" ng-if="!project" translatable="yes">New project</h4>
+        <h4 class="modal-title" ng-if="!project" translatable="yes">
```
Comment (petervo) [inline pkg/kubernetes/views/project-modify.html]: same as above translatable elements can't have children

File: pkg/kubernetes/views/project-modify.html [hunk 9]
```diff
@@ -1,13 +1,23 @@
...
+          </a>
+        </h4>
         <h3 class="modal-title" ng-if="project" translatable="yes">Change project</h3>
     </div>
     <div class="modal-body">
         <table class="cockpit-form-table">
             <tr>
                 <td class="top">
-                    <label class="control-label" for="project-new-name" translatable="yes">Name</label>
+                    <label class="control-label" for="project-new-name" translatable="yes">
```
Comment (petervo) [inline pkg/kubernetes/views/project-modify.html]: ditto

File: pkg/kubernetes/views/registry-dashboard-page.html [hunk 10]
```diff
@@ -14,7 +14,12 @@ <h1 translatable="yes">Welcome to the Image Registry</h1>
 <div class="container-cards-pf dashboard-cards" ng-show="filter.namespaces().length">
     <div class="card-pf dashboard-images">
         <div class="card-pf-heading">
-            <h2 class="card-pf-title" translatable="yes">Images by project</h2>
+            <h2 class="pull-left card-pf-title" translatable="yes">Images by project</h2>
+            <h2 class="pull-right card-pf-link-with-icon new-project-link">
+              <a ng-click="createProject()">
```
Comment (petervo) [inline pkg/kubernetes/views/registry-dashboard-page.html]: Is the present on the scope?

File: pkg/kubernetes/views/project-modify.html [hunk 11]
```diff
@@ -1,13 +1,23 @@
...
+                      <a tabindex="0" role="button" popover-trigger="focus"
+                          popover="The project name is used as the repository name when referencing images, registry/project/image:[tag]"<span class="fa fa-lg fa-info-circle"></span>
```
Comment (petervo) [inline pkg/kubernetes/views/project-modify.html]: the closing > for the a tag is missing. Prevents building.

File: pkg/kubernetes/views/project-modify.html [hunk 12]
```diff
@@ -1,13 +1,23 @@
 <modal-dialog>
     <div class="modal-header">
-        <h4 class="modal-title" ng-if="!project" translatable="yes">New project</h4>
+        <h4 class="modal-title" ng-if="!project" translatable="yes">
+          New project
+          <a tabindex="0" role="button" popover-trigger="focus"
+              popover="A namespace for images"<span class="fa fa-lg fa-info-circle"></span>
```
Comment (petervo) [inline pkg/kubernetes/views/project-modify.html]: the closing > for the a tag is missing. Prevents building.

File: pkg/kubernetes/views/registry-dashboard-page.html [hunk 13]
```diff
@@ -14,7 +14,12 @@ <h1 translatable="yes">Welcome to the Image Registry</h1>
...
+                <i class="pficon pficon-add-circle-o"></i>
+              </a>
+            <h2>
```
Comment (petervo) [inline pkg/kubernetes/views/registry-dashboard-page.html]: should be </h2>

**Comment by andreasn** [comment]
Found the popover placement code that worked.
`popover-placement="bottom"`
makes the popover expand downwards instead of up, and avoids getting cut by the Cockpit top bar.

**Comment by andreasn** [comment]
` popover-placement="left"`
and
` popover-placement="right"`
are also valid arguments

**Comment by andreasn** [comment]
>For some reason, I get stuck on this when I run make:
>GEN pkg/kubernetes/kubernetes-templates.js
>No errors or anything. It just gets stuck.

Got it building by closing the tags @petervo suggested above.

**Comment by andreasn** [comment]
The icon is a bit off vertically, so something like this is needed for cockpit.css
```
.cockpit-form-table label span.fa-info-circle {
    vertical-align: initial;
}
```

**Comment by andreasn** [comment]
I wonder if the helper text would work better inside the text field like this:

screenshot from 2016-04-26 13-14-22

will need this styling to the a:
```
position: absolute;
right: 18px;
```

**Comment by petervo** [comment]
Replaced by #4300

**Pull request closed without merging**